tests/Feature/Commands/TweetAboutUpcomingStreamsCommandTest.php

```diff
@@ -0,0 +1,143 @@
+
+        // Assert
+        $this->twitterFake->assertLastTweetMessageWas($expectedStatus);
+    }
+
+    /** @test */
+    public function it_adds_twitter_handle_to_streams_connected_to_a_channel(): void
+    {
+        // Arrange
+        $stream = Stream::factory()
+            ->upcoming()
+            ->for(Channel::factory()->create(['twitter_handle' => '@twitterUser']))
+            ->create(['scheduled_start_time' => now()->addMinutes(5)]);
+
+        $expectedStatus = "🔴 A new stream by @twitterUser is about to start: $stream->title. Join now!\n{$stream->url()}";
+
+        // Act
+        $this->artisan(TweetAboutUpcomingStreamsCommand::class);
+
+        // Assert
+        $this->twitterFake->assertLastTweetMessageWas($expectedStatus);
+    }
+
+    /** @test */
+    public function it_works_without_missing_twitter_handle_on_connected_channel(): void
+    {
+        // Arrange
+        $stream = Stream::factory()
+            ->upcoming()
+            ->for(Channel::factory())
+            ->create(['scheduled_start_time' => now()->addMinutes(5)]);
+
+        $expectedStatus = "🔴 A new stream is about to start: $stream->title. Join now!\n{$stream->url()}";
+
+        // Act
+        $this->artisan(TweetAboutUpcomingStreamsCommand::class);
+
+        // Assert
+        $this->twitterFake->assertLastTweetMessageWas($expectedStatus);
+    }
+
+    /** @test */
+    public function it_correctly_sets_announcement_tweeted_at_timestamp(): void
+    {
+        // Arrange
+        Carbon::setTestNow(now());
+        $upcomingStreamToTweet = Stream::factory()->upcoming()->create(['scheduled_start_time' => now()->addMinutes(5)]);
+        $upcomingStreamNotToTweet = Stream::factory()->upcoming()->create(['scheduled_start_time' => now()->addMinutes(6)]);
+        $liveStreamNotToTweet = Stream::factory()->live()->create();
+
+        // Assert
+        $this->assertNull($upcomingStreamToTweet->announcement_tweeted_at);
+        $this->assertNull($upcomingStreamNotToTweet->announcement_tweeted_at);
+        $this->assertNull($liveStreamNotToTweet->announcement_tweeted_at);
+
+        // Act
+        $this->artisan(TweetAboutUpcomingStreamsCommand::class)
+            ->expectsOutput('1 tweets sent')
+            ->assertExitCode(0);
+
+        // Assert
+        tap($upcomingStreamToTweet->fresh(), function($upcomingStreamToTweet) {
+            $this->assertNotNull($upcomingStreamToTweet->announcement_tweeted_at);
+            $this->assertInstanceOf(Carbon::class, $upcomingStreamToTweet->announcement_tweeted_at);
+            $this->assertSame((string) now(), (string) $upcomingStreamToTweet->announcement_tweeted_at);
+        });
+
+        $this->assertNull($upcomingStreamNotToTweet->refresh()->announcement_tweeted_at);
+        $this->assertNull($liveStreamNotToTweet->refresh()->announcement_tweeted_at);
+    }
+
+    /** @test */
+    public function it_does_not_send_a_tweet_if_the_live_tweet_is_sent(): void
```

Comment:
@christophrumpel i have added this test (and corresponding check in the job) to make sure we don't tweet about an upcoming stream if the tweet for the "stream is live" is already sent. This should not happen but in the odd case the job is running long or something we should not tweet it out i think?

Reply:
yeah good catch 👍